src/ycp.i

```diff
@@ -0,0 +1,322 @@
+        return (*($self))->isList();
+    }
+    bool isTerm() {
+        return (*($self))->isTerm();
+    }
+    bool isMap() {
+        return (*($self))->isMap();
+    }
+    bool isCode() {
+        return (*($self))->isCode();
+    }
+    bool isBreak() {
+        return (*($self))->isBreak();
+    }
+    bool isReturn() {
+        return (*($self))->isReturn();
+    }
+    bool isEntry() {
+        return (*($self))->isEntry();
+    }
+    bool isReference() {
+        return (*($self))->isReference();
+    }
+    YCPVoid asVoid() {
+        return (*($self))->asVoid();
+    }
+    YCPBoolean asBoolean() {
+        return (*($self))->asBoolean();
+    }
+    YCPInteger asInteger() {
+        return (*($self))->asInteger();
+    }
+    YCPFloat asFloat() {
+        return (*($self))->asFloat();
+    }
+    YCPString asString() {
+        return (*($self))->asString();
+    }
+    YCPByteblock asByteblock() {
+        return (*($self))->asByteblock();
+    }
+    YCPPath asPath() {
+        return (*($self))->asPath();
+    }
+    YCPSymbol asSymbol() {
+        return (*($self))->asSymbol();
+    }
+    YCPList asList() {
+        return (*($self))->asList();
+    }
+    YCPTerm asTerm() {
+        return (*($self))->asTerm();
+    }
+    YCPMap asMap() {
+        return (*($self))->asMap();
+    }
+    YCPCode asCode() {
+        return (*($self))->asCode();
+    }
+    YCPEntry asEntry() {
+        return (*($self))->asEntry();
+    }
+    YCPReference asReference() {
+        return (*($self))->asReference();
+    }
+    bool equal (const YCPValue&v) {
+        return (*($self))->equal(v);
+    }
+    YCPOrder compare(const YCPValue &v, bool rl = false) {
+        return (*($self))->compare(v, rl);
+    }
+}
+
+%ignore YCPTermRep;
+%rename(Term) YCPTerm;
+%include <ycp/YCPTerm.h>
+%extend YCPTerm {
+    YCPOrder compare(const YCPTerm &v) {
+        return (*($self))->compare(v);
+    }
+    string toString() {
+        return (*($self))->toString();
+    }
+}
+
+%ignore YCPListRep;
+%rename(List) YCPList;
+%include <ycp/YCPList.h>
+%extend YCPList {
+    YCPOrder compare(const YCPList &v) {
+        return (*($self))->compare(v);
+    }
+    string toString() {
+        return (*($self))->toString();
+    }
+}
+
+%feature("valuewrapper") YCPString;
+class YCPString;
+%ignore YCPStringRep;
+%rename(String) YCPString;
+%include <ycp/YCPString.h>
+%extend YCPString {
+    bool isAscii() {
+        return (*($self))->isAscii();
+    }
+    const string& value() {
+        return (*($self))->value();
+    }
+    wstring wvalue() {
+        return (*($self))->wvalue();
+    }
+    YCPOrder compare(const YCPString &v, bool rl = false) {
+        return (*($self))->compare(v, rl);
+    }
+    const char *value_cstr() {
+        return (*($self))->value_cstr();
+    }
+    string toString() {
+        return (*($self))->toString();
+    }
+}
+
+%ignore YCPSymbolRep;
+%rename(Symbol) YCPSymbol;
+%include <ycp/YCPSymbol.h>
+%extend YCPSymbol {
```

Comment:
We wrap YCPSymbol (and rename to Symbol) here so that it can be manually created, in case we need to manually create a Symbol. This is exactly how symbol's were handled in the old python bindings (which preserves backwards compatibility also).

Reply:
ok, you will see in your code ;)